public/classes/textile.ts

```diff
@@ -1,22 +1,22 @@
-import { RowInfo } from "../model/types";
+import { Color, TextileState, WarpConfig } from "../model/types";
 
 export interface TextileOptions {
   cols: number;
   name: string;
 }
 
-export class Textile{
+export class Textile implements TextileState{
```

Comment:
This pattern of `Class implements ClassState` could help us pass objects around without worrying about circular import issues. I'm not sold on the idea, so let me know what you think!